gobblin-runtime/src/main/java/gobblin/runtime/MysqlDatasetStateStore.java

```diff
@@ -64,12 +94,57 @@ public MysqlDatasetStateStore(DataSource dataSource, String stateStoreTableName,
    * @throws IOException if there's something wrong reading the {@link JobState.DatasetState}s
    */
   public Map<String, JobState.DatasetState> getLatestDatasetStatesByUrns(String jobName) throws IOException {
-    List<JobState.DatasetState> previousDatasetStates =
-        getAll(jobName, "%" + CURRENT_DATASET_STATE_FILE_SUFFIX + DATASET_STATE_STORE_TABLE_SUFFIX, true);
+    final String storeName = jobName;
```

Comment:
This exposes the querying logic to the MysqlDatasetStateStore and also results in duplication of logic in the result processor.